mesonbuild/modules/dub.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2013-2017 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains the detection logic for external dependencies that
+# are UI-related.
+
+import json
+import os
+
+from . import ExtensionModule
+
+from .. import mlog
+
+from ..mesonlib import (
+    Popen_safe, MesonException
+)
+
+from ..dependencies.base import (
+    ExternalProgram, DubDependency
+)
+
+from ..interpreter import DependencyHolder
+
+class DubModule(ExtensionModule):
+    class_dubbin = None
+
+    def __init__(self, interpreter):
+        super().__init__(interpreter)
+        self.snippets.add('generate_dub_file')
+
+        if DubModule.class_dubbin is None:
+            self.dubbin = DubDependency.class_dubbin
+            DubModule.class_dubbin = self.dubbin
+        else:
+            self.dubbin = DubModule.class_dubbin
+
+        if DubModule.class_dubbin is None:
+            self.dubbin = self.check_dub()
+            DubModule.class_dubbin = self.dubbin
+        else:
+            self.dubbin = DubModule.class_dubbin
+
+        if not self.dubbin:
+            if not self.dubbin:
+                raise MesonException('DUB not found.')
+
+    def generate_dub_file(self, interpreter, state, args, kwargs):
+        if len(args) < 2:
+            raise MesonException('Missing arguments')
+
+        no_publishing = False
+        if 'no_publishing' in kwargs:
+            if not kwargs.get('no_publishing', []):
+                no_publishing = True
+
+        warn_publishing = ['description', 'license']
+        for arg in warn_publishing:
+            if arg not in kwargs and not no_publishing:
+                mlog.warning('Without', mlog.bold(arg), 'the DUB package can\'t be published')
+
+        config = {
+            'name': args[0]
+        }
+
+        config_path = os.path.join(args[1], 'dub.json')
+        if os.path.exists(config_path):
+            with open(config_path, 'r', encoding='utf8') as ofile:
+                try:
+                    config = json.load(ofile)
+                except:
```

Comment:
__[Flake8]__

[E722] do not use bare except'


<!-- Comment by [SideCI](https://sideci.com) -->